diff --git a/src/plugins/avfoundation/camera/avfcamerautility.mm b/src/plugins/avfoundation/camera/avfcamerautility.mm
index 25ccc4b01..4a7cb3b13 100644
--- a/src/plugins/avfoundation/camera/avfcamerautility.mm
+++ b/src/plugins/avfoundation/camera/avfcamerautility.mm
@@ -101,7 +101,7 @@ bool operator() (AVCaptureDeviceFormat *f1, AVCaptureDeviceFormat *f2)const
     }
 };
 
-struct FormatHasNoFPSRange : std::unary_function<AVCaptureDeviceFormat *, bool>
+struct FormatHasNoFPSRange : std::__unary_function<AVCaptureDeviceFormat *, bool>
 {
     bool operator() (AVCaptureDeviceFormat *format)
     {
